### PR TITLE
[BANKCON-14531] Update Instant Debits flow to call `/share` endpoint for Panther payments

### PR DESCRIPTION
**Pull request by mats-stripe**
Android equivalent: https://github.com/stripe/stripe-android/pull/9308

## Summary

This updates the Instant Debits flow to call the `/consumers/payment_details/share` endpoint instead of the `/payment_methods` endpoint for Panther payments. This endpoint, along with the `expected_payment_method_type: "card"` parameter will indicate whether or not we should "Panther-ize" the payment method response. Lots more details in the [Panther eng plan](https://docs.google.com/document/d/1ErJVA3lLvNspPe3A8uYP9feK8Yvfq-Sw5aatNIvlGxk/edit?usp=sharing)!

## Motivation

BANKCON-14531

## Testing

With some minor changes to the mobile payment elements (PR coming next!) we can now see the full e2e flow in action!

https://github.com/user-attachments/assets/426fee87-7c93-4f0f-9f93-ad29f1d762bd

## Changelog

N/a

### CODE REVIEW

File: StripeCore/StripeCore/Source/Helpers/Async.swift [hunk 1]
```diff
@@ -114,9 +114,10 @@ import Foundation
     }
 
     public func transformed<T>(
+        on queue: DispatchQueue = .main,
         with closure: @escaping (Value) throws -> T
     ) -> Future<T> {
-         chained { value in
+        chained(on: queue) { value in
```
Comment (mats-stripe) [inline StripeCore/StripeCore/Source/Helpers/Async.swift:120]: Drive-by cleanup of this [recently merged method](https://github.com/stripe/stripe-ios/pull/4051)!

File: StripeFinancialConnections/StripeFinancialConnections/Source/API Bindings/FinancialConnectionsAPIClient.swift [hunk 2]
```diff
@@ -944,6 +950,27 @@ extension FinancialConnectionsAPIClient: FinancialConnectionsAPI {
...
+    ) -> Future<FinancialConnectionsSharePaymentDetails> {
+        let parameters: [String: Any] = [
+            "request_surface": requestSurface,
+            "id": paymentDetailsId,
+            "credentials": [
+                "consumer_session_client_secret": consumerSessionClientSecret
+            ],
+            "expected_payment_method_type": expectedPaymentMethodType,
+            "expand": ["payment_method"],
+        ]
```
Comment (mats-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/API Bindings/FinancialConnectionsAPIClient.swift:966]: Details: https://docs.google.com/document/d/1ErJVA3lLvNspPe3A8uYP9feK8Yvfq-Sw5aatNIvlGxk/edit#bookmark=id.oljqxd4uzx7t

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/NativeFlowController.swift [hunk 3]
```diff
@@ -491,32 +492,48 @@ extension NativeFlowController {
...
+                    paymentDetailsId: paymentDetails.redactedPaymentDetails.id,
+                    expectedPaymentMethodType: "card"
+                )
+                .transformed { $0 as PaymentMethodIDProvider }
+            } else {
+                return self.dataManager.createPaymentMethod(
+                    consumerSessionClientSecret: consumerSession.clientSecret,
+                    paymentDetailsId: paymentDetails.redactedPaymentDetails.id
+                )
+                .transformed { $0 as PaymentMethodIDProvider }
```
Comment (mats-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Native/NativeFlowController.swift:535]: This is where we get to use that new type-erasure API!

**Review by tillh-stripe — APPROVED** [review]
Looking good from a Link Card Brand perspective!
